test/upgrade/main.go

```diff
@@ -104,50 +136,164 @@ func findK8sVersion() string {
 		log.Fatal("Could not create the kubernetes api clientset", err)
 	}
 
-	ctx := context.Background()
+	// Wait to get pod and node as these may take a while to start on a new Autopilot cluster.
+	var pod *v1.Pod
+	err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 7*time.Minute, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
By definition `findK8sVersion` is running in a container in a pod, but somehow `pod, err = kubeClient.CoreV1().Pods(POD_NAMESPACE).Get(ctx, POD_NAME, metav1.GetOptions{})` runs into errors for ~3 min as the pod is trying to schedule itself onto a node. How this manages to run without yet being scheduled to a Node, and how it can be on a pod and not be able to gets its name I don't quite understand, but that's what the logs are showing. So the PollUntilContextTimeout waits until the kubeClient is able to successfully get the pod name, or if it can't get the pod name after 7 min it errors out.